File: test/benchmark/Math/python/factors/factors.py
import sys
import time
from datetime import datetime

def factors(n):
    result = []
    for i in range(1, n + 1):
        if n % i == 0:
            result.append(i)
    return result

n = int(sys.argv[1])
start_time = time.time_ns()
print(f"Start time: {start_time} ns")
print(factors(n))
end_time = time.time_ns()
execution_time = (end_time - start_time) / 1e9
print(f"Execution time: {execution_time:.9f} seconds")

File: test/benchmark/Math/c/factors/factors.cpp
#include <cstdint>
#include <cstdio>
#include <cstdlib>   // for std::atoi
#include <chrono>
#include <inttypes.h> // for PRId64
using namespace std;

int main(int argc, char* argv[]) {
    auto start_time = chrono::system_clock::now();
    int64_t ns = chrono::duration_cast<chrono::nanoseconds>(start_time.time_since_epoch()).count();

    printf("Start time: %" PRId64 " ns\n", ns);

    if (argc < 2) {
        printf("Usage: %s <number>\n", argv[0]);
        return 1;
    }

    int32_t n = std::atoi(argv[1]);  // atoi is in <cstdlib>
    int32_t count = 0;

    for (int32_t i = 1; i <= n; ++i) {
        if (n % i == 0) {
            printf("%d\n", i);
            count++;
        }
    }
    printf("Total factors: %d\n", count);

    auto end_time = chrono::system_clock::now();
    int64_t duration_ns = chrono::duration_cast<chrono::nanoseconds>(end_time - start_time).count();
    double duration_s = duration_ns / 1e9;

    printf("Execution time: %f seconds\n", duration_s);
    return 0;
}
